activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -193,6 +205,80 @@ def wait_poll(timeout)
         end
       end
 
+      # Adds the ability to turn a basic fair FIFO queue into one
+      # biased to some thread.
+      module BiasableQueue # :nodoc:
+        class BiasedConditionVariable
```

Comment:
@tenderlove Absolutely. @pitr-ch has done excellent work creating an enhanced synchronization layer and is currently working on an [improved condition variable](https://github.com/ruby-concurrency/concurrent-ruby/blob/synchronization/lib/concurrent/synchronization/condition.rb). I don't see why we couldn't add this. @pitr-ch: Thoughts?